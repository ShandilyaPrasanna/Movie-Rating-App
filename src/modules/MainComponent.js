import React, { Component } from "react";



export class MainComponent extends Component {

  render() {
    return ("HELLO");
  }
}


export default (MainComponent);
