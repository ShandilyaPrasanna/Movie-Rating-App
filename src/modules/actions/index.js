import MovieAppApi from "../api/movieApi";
import Constants from "../constants/movieAppConstants";

export function getMoviesAction(endPoint) {
  return function(dispatch) {
    MovieAppApi.fetchMovies(endPoint)
      .then(result => {
        dispatch({ type: Constants.actions.GET_MOVIES, payload: result });
      })
      .catch(error => {
        console.error("Something went wrong", error);
        dispatch(isLoading(false));
      });
  };
}

export function isLoading(bool) {
  return { type: Constants.actions.IS_LOADING, payload: bool };
}
