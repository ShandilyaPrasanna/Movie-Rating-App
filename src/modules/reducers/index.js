import { combineReducers } from "redux";
import movieAppReducer from "./movieAppReducer";

export const reducers = combineReducers({
  app: movieAppReducer
});
