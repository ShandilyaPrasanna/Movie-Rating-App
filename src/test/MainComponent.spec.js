import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import toJson from "enzyme-to-json";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { MainComponent, mapStateToProps } from "../modules/MainComponent";

const state = {
  app: {
    loading: true,
    images: [],
    pagination: {}
  }
};

configure({ adapter: new Adapter() });
describe("MainComponent Test", () => {
  let tree;
  const getMoviesAction = jest.fn();
  const isLoading = jest.fn();
  const props = {
    getMoviesAction,
    isLoading,
    loading: true
  };
  beforeEach(() => {
    tree = shallow(<MainComponent {...props} />);
  });

  it("renders MainComponent ", () => {
    expect(toJson(tree)).toMatchSnapshot();
  });

  it("mapStateToProps ", () => {
    expect(mapStateToProps({ app: { loading: true } }).loading).toBe(true);
  });
});
