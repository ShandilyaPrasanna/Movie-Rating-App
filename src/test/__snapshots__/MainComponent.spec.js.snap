// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent Test renders MainComponent  1`] = `
<WithStyles(ForwardRef(Container))
  maxWidth="lg"
>
  <div
    className="spinner"
  />
  <Header />
  <WithStyles(undefined) />
  <WithStyles(undefined) />
  <Footer />
</WithStyles(ForwardRef(Container))>
`;
